### ISSUE TITLE
fatal: [default]: FAILED! => {"changed": false, "failed": true, "msg": "__init__() got an unexpected keyword argument 'validate_certs'"}

### ISSUE THREAD

**Issue by zacho19** [ansible/ansible-modules-extras#2449]
ISSUE TYPE
Bug Report
ANSIBLE VERSION

```
ansible 2.1.0.0
  config file =
  configured module search path = Default w/o overrides
```

CONFIGURATION

None

OS / ENVIRONMENT

Running el captain 10.11.5. Managing ubuntu/trusty64 vagrant box

SUMMARY

I want to run vault in a vagrant box using ansible. Trying to start the vault server as a service with conusl, however it throws {"changed": false, "failed": true, "msg": "init() got an unexpected keyword argument 'validate_certs'"} error. I have tried adding validate_certs: false but it doesn't change anything. There also doesn't appear to be a module to specify the config file.

STEPS TO REPRODUCE

Simple vagrantfile with ubuntu/trusty64 base box, and ansible provisioning
vagrant up

```

---
- hosts: all
  become: yes
  become_method: sudo
  tasks:
#aquire vault and config files
  - apt: name=unzip state=present
  - unarchive: src=https://releases.hashicorp.com/vault/0.5.3/vault_0.5.3_linux_amd64.zip dest=/usr/local/bin copy=no
  - name: Transfer vault config
    copy: src=./config.hcl dest=/home/vagrant

#consul dependencies
  - name: Download pip.
    get_url: url=https://bootstrap.pypa.io/get-pip.py dest=/usr/local/bin
  - name: Install pip.
    command: python /usr/local/bin/get-pip.py
  - name: Install python-consul module
    pip: name=python-consul state=present

#vault as a service
  - name: register vault service with the local consul agent
    consul:
      service_name: vault
      service_port: 80
#    validate_certs: false
```

config.hcl

```
backend "consul" {
  address = "127.0.0.1:8500"
  path = "vault"
  advertise_addr = "http://127.0.0.1:8500"
}

listener "tcp" {
  address = "127.0.0.1:8200"
  tls_disable = 1
}

disable_mlock = true
```

ACTUAL RESULTS

```
fatal: [default]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"check_host": null, "check_id": null, "check_name": null, "check_node": null, "host": "localhost", "http": null, "interval": null, "notes": null, "port": 8500, "scheme": "http", "script": null, "service_address": "localhost", "service_id": null, "service_name": "vault", "service_port": 80, "state": "present", "tags": null, "timeout": null, "token": null, "ttl": null, "validate_certs": true}, "module_name": "consul"}, "msg": "init() got an unexpected keyword argument 'validate_certs'"}
```

```
<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: vagrant
<127.0.0.1> SSH: EXEC ssh -C -vvv -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2222 -o 'IdentityFile="/Users/zlanger/workspace/vagrant_files2/.vagrant/machines/default/virtualbox/private_key"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 -o ControlPath=/Users/zlanger/.ansible/cp/ansible-ssh-%h-%p-%r 127.0.0.1 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1466184741.78-47511526590053 `" && echo ansible-tmp-1466184741.78-47511526590053="` echo $HOME/.ansible/tmp/ansible-tmp-1466184741.78-47511526590053 `" ) && sleep 0'"'"''
<127.0.0.1> PUT /var/folders/73/bmxj72mj6bgffs_p3sjtgw740000gp/T/tmpdJpJR6 TO /home/vagrant/.ansible/tmp/ansible-tmp-1466184741.78-47511526590053/consul
<127.0.0.1> SSH: EXEC sftp -b - -C -vvv -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2222 -o 'IdentityFile="/Users/zlanger/workspace/vagrant_files2/.vagrant/machines/default/virtualbox/private_key"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 -o ControlPath=/Users/zlanger/.ansible/cp/ansible-ssh-%h-%p-%r '[127.0.0.1]'
<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: vagrant
<127.0.0.1> SSH: EXEC ssh -C -vvv -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2222 -o 'IdentityFile="/Users/zlanger/workspace/vagrant_files2/.vagrant/machines/default/virtualbox/private_key"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 -o ControlPath=/Users/zlanger/.ansible/cp/ansible-ssh-%h-%p-%r -tt 127.0.0.1 '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-jjqvevzvyvrfpfxifzosqzwnwalbhrqe; LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1466184741.78-47511526590053/consul; rm -rf "/home/vagrant/.ansible/tmp/ansible-tmp-1466184741.78-47511526590053/" > /dev/null 2>&1'"'"'"'"'"'"'"'"' && sleep 0'"'"''
```

**Comment by alikins** [comment]
This appears to be the same issue as https://github.com/ansible/ansible-modules-extras/issues/2114

A pr with the fix (https://github.com/ansible/ansible-modules-extras/pull/2194) has been merged into devel branch at e388fb4006e1d8f7828527a02c445bc633b5b367

and into the stable-2.1 branch at d7ea280e5081fccb9d82ba715c88c2e3ad361446

So the fix for this will be in the next 2.1.x release.

**Comment by alikins** [comment]
Since this is a dupe of #2114, and that is the issue number referenced in the fixing commits and backports, I'm going to close this issue.
